T19387 - only show failed builds when landing on page

alexandrasp opened the pull request:
remove all session storage in order to avoid
bugs and to only show failed builds when there is any
fail on build list.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker reviewed (commented):
> remove all session storage in order to avoid
> bugs and to only show failed builds when there is any
> fail on build list.

The commit title should also show that this is about removing session storage, not just showing failed builds by default.

gctucker reviewed (changes requested):
Well it's one thing to disable session storage, the problem is that the issue is still visible:
https://staging.kernelci.org/build/kernelci/branch/staging.kernelci.org/kernel/staging-20200226.0/

gctucker commented:
If you open the view above, you can see that the "Failed" button is active but both successful and failed builds are still visible in the list.  Only failed builds should be visible instead.

alexandrasp commented:
> If you open the view above, you can see that the "Failed" button is active but both successful and failed builds are still visible in the list. Only failed builds should be visible instead.

yes, I see. I will check this again, now I have the same problem locally.

Commit:
fix-failed-problem-landing-on-page-builds

Include a promisse to ensure that a
table with build information will
exists before to performing the filter.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker commented:
This PR caused the staging update to fail in this ansible step:
```
TASK [configure-nginx : Build static assets]
```

alexandrasp commented:
> This PR caused the staging update to fail in this ansible step:
> 
> ```
> TASK [configure-nginx : Build static assets]
> ```

Ok, After finish main test view I will back to this problem.

alexandrasp commented:
@gctucker could you remove staging-skip ? I removed promise and now the code just wait for 1s for data to arrive and table to be built.

gctucker commented on app/dashboard/static/js/app/view-builds-job-branch-kernel.2020.2.1.js:
```diff
@@ -61,13 +61,13 @@ require([
                 value.addEventListener(
                     'click', commonBtns.showHideElements, true);
-        });
+            });
```
Unnecessary whitespace noise

gctucker commented:
OK it's on staging now, the ansible issue is gone.

gctucker reviewed (changes requested):
Well that appears to "work" but it's not an acceptable solution to just wait for one second every time this page is loaded and hope the results have been received in the meantime.

alexandrasp commented:
> Well that appears to "work" but it's not an acceptable solution to just wait for one second every time this page is loaded and hope the results have been received in the meantime.

I see and agree with you! It was good to see if it works on staging, now I will tune this solution.

gctucker commented:
I think I've fixed this problem with #91.
